src/python/pants/backend/codegen/protobuf/scala/dependency_inference.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from dataclasses import dataclass
+
+from pants.backend.codegen.protobuf.scala.subsystem import ScalaPBSubsystem
+from pants.backend.codegen.protobuf.target_types import ProtobufDependenciesField
+from pants.backend.scala.subsystems.scala import ScalaSubsystem
+from pants.build_graph.address import Address
+from pants.engine.internals.selectors import Get
+from pants.engine.rules import collect_rules, rule
+from pants.engine.target import InjectDependenciesRequest, InjectedDependencies, WrappedTarget
+from pants.engine.unions import UnionRule
+from pants.jvm.dependency_inference.artifact_mapper import AllJvmArtifactTargets
+from pants.jvm.resolve.common import ArtifactRequirement, Coordinate
+from pants.jvm.subsystems import JvmSubsystem
+from pants.jvm.target_types import JvmResolveField
+from pants.util.docutil import bin_name
+
+_SCALAPB_RUNTIME_GROUP = "com.thesamet.scalapb"
+_SCALAPB_RUNTIME_ARTIFACT = "scalapb-runtime"
+
+
+class InjectScalaPBRuntimeDependencyRequest(InjectDependenciesRequest):
+    inject_for = ProtobufDependenciesField
+
+
+@dataclass(frozen=True)
+class ScalaPBRuntimeForResolveRequest:
+    resolve_name: str
+
+
+@dataclass(frozen=True)
+class ScalaPBRuntimeForResolve:
+    address: Address
+
+
+@rule
+async def resolve_scalapb_runtime_for_resolve(
+    request: ScalaPBRuntimeForResolveRequest,
+    jvm_artifact_targets: AllJvmArtifactTargets,
+    jvm: JvmSubsystem,
+    scala_subsystem: ScalaSubsystem,
+    scalapb: ScalaPBSubsystem,
+) -> ScalaPBRuntimeForResolve:
+    scala_version = scala_subsystem.version_for_resolve(request.resolve_name)
+    # TODO: Does not handle Scala 3 suffix which is just `_3`.
+    scala_binary_version = ".".join(scala_version.split(".")[0:2])
```

Comment:
Assuming this is a semver `x.y.z` string, a better way to write this would be 

```
scala_binary_version, _, _ = scala_version.rpartition(".")
```


Example in repl:

```
>>> "2.11.6".rpartition(".")
('2.11', '.', '6')
```

Reply:
This should probably be refactored into a function to compute the artifact suffix since this naive logic doesn't handle the Scala 3 suffix logic.

Reply:
Going with `rpartition` for now and an expanded TODO comment.